text/0141-linking-traces.md

```diff
@@ -0,0 +1,361 @@
+
+## Additional Use Cases
+
+In addition, besides inter-trace links, we might also be interested in linking spans within one trace, for example to link spans that are related to each other but not in a direct (parent-child or linear) hierarchy. While this is not the main objective of the RFC, our proposed solution would also facilitate this linkage. Some concrete examples (but out of scope):
+- (Async) data flow operations (e.g. RxJS pipes) where we would be able to link individual operations.
+- Websocket spans where we could link individual web socket messages
+- More generally, whenever we want to "casually" establish a relationship between spans that cannot or should not be achieved via a hard relationship (traceId or parent spanId)
+
+# Concrete Goals of achieving linked traces
+
+Once this RFC is accepted, we have settled on all important details needed to implement connected traces.
+This means, we have all the necessary information to support the following concrete goals:
+
+TODO: List concrete product/UI goals
+
+- SDKs can add support for adding links to spans
+- We can ingest and store span links
+- Frontend SDKs can automatically add span links so that the current trace root span links back to the previous trace root span
+
+
+## Secondary goals
+
+- We can shorten the lifetime of frontend traces because we're now able to link to the previous trace instead. This is something we'll do but it's a follow-up and can only be addressed once the primary goals are achieved. 
+
+## Out of scope/Non-Goals
+
+- Querying across linked traces: Given the "linked list" nature of one root span linking back to its previous root span, we accept that obtaining a complete list is an expensive query operation. Therefore, we consider use cases that require a complete list of all linked traces out of scope. We can address this as a follow-up if technically possible but we're aware that this is likely a sub-optimal data structure for such queries. 
+- Linking transactions started during the rendering of statically generated HTML pages and pageload transactions. A span link is ideal here because we don't want the hard link of a trace id. However, a casual link would still increase context and completeness. Wwe might do it but not at this point as it's a follow-up item.
```

Comment:
```suggestion
- Linking transactions started during the rendering of statically generated HTML pages and pageload transactions. A span link is ideal here because we don't want the hard link of a trace id. However, a casual link would still increase context and completeness. We might do it but not at this point as it's a follow-up item.
```